csource/csource_test.go

```diff
@@ -16,36 +16,88 @@ import (
 
 func initTest(t *testing.T) (rand.Source, int) {
 	t.Parallel()
-	iters := 10
-	if testing.Short() {
-		iters = 1
-	}
+	iters := 1
 	seed := int64(time.Now().UnixNano())
 	rs := rand.NewSource(seed)
 	t.Logf("seed=%v", seed)
 	return rs, iters
 }
 
+func allOptionsSingle() []Options {
+	var options []Options
+	var opt Options
+	for _, opt.Threaded = range []bool{false, true} {
```

Comment:
Is it possible to not duplicate all options here and in allOptionsPermutations somehow?